File: Python/BaekJoon/2884_알람 시계.py
# 방법 1
hour, min = map(int, input().split())

if min < 45:
  if hour == 0:
    hour = 23
    min += 60
  else:
    hour -= 1
    min += 60

print(hour, min-45)

# 방법 2
# hour, minute = map(int, input().split())

# if minute > 44:
#     print(hour, minute-45)
# elif minute <= 44 and hour >= 1:
#     print(hour-1, minute+15)
# else:
#     print(23, minute+15)

File: Java/BaekJoon/2884_알람시계.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine(), " ");

        int hour = Integer.parseInt(st.nextToken());
        int min = Integer.parseInt(st.nextToken());

        br.close();

        StringBuilder sb = new StringBuilder();

        if (min < 45) {
            if (hour == 0) {
                hour = 23;
                sb.append(hour).append(" ");
            } else {
                hour -= 1;
                sb.append(hour).append(" ");
            }
            sb.append(min = 60 - (45 - min));
        } else {
            sb.append(hour).append(" ").append(min - 45);
        }

        System.out.println(sb);
    }
}